src/Common/Admin/Help_Hub/Tab_Builder.php

```diff
@@ -0,0 +1,225 @@
+	/**
+	 * The tab target.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private string $target;
+
+	/**
+	 * The tab CSS class.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private string $class = '';
+
+	/**
+	 * The tab label.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private string $label;
+
+	/**
+	 * The tab ID.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private string $id;
+
+	/**
+	 * The tab template.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private string $template;
+
+	/**
+	 * The tab arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string, mixed>
+	 */
+	private array $args = [];
+
+	/**
+	 * Static storage for all built tabs.
+	 *
+	 * @since TBD
+	 *
+	 * @var array<string, array>
+	 */
+	private static array $tabs = [];
+
+	/**
+	 * Create a new tab instance.
+	 *
+	 * @since TBD
+	 *
+	 * @param string               $target   The tab target.
+	 * @param string               $label    The tab label.
+	 * @param string               $id       The tab ID.
+	 * @param string               $template The tab template.
+	 * @param array<string, mixed> $args     Optional arguments for the tab.
+	 *
+	 * @return static
+	 */
+	public static function make( string $target, string $label, string $id, string $template, array $args = [] ): self {
+		$instance           = new self();
+		$instance->target   = $target;
+		$instance->label    = $label;
+		$instance->id       = $id;
+		$instance->template = $template;
+		$instance->args     = $args;
+
+		return $instance;
+	}
+
+	/**
+	 * Set the tab CSS class.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $class_name The tab CSS class.
+	 *
+	 * @return $this
+	 */
+	public function set_class( string $class_name ): self {
+		$this->class = $class_name;
+
+		return $this;
+	}
+
+	/**
+	 * Set the tab arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string, mixed> $args The tab arguments.
+	 *
+	 * @return $this
+	 */
+	public function set_arguments( array $args ): self {
+		$this->args = $args;
+
+		return $this;
+	}
+
+	/**
+	 * Get the tab arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @return array<string, mixed> The tab arguments.
+	 */
+	public function get_arguments(): array {
+		return $this->args;
+	}
+
+	/**
+	 * Build the tab array and store it.
+	 *
+	 * @since TBD
+	 *
+	 * @return array{
+	 *     target: string,
+	 *     class: string,
+	 *     label: string,
+	 *     id: string,
+	 *     template: string,
+	 *     args: array<string, mixed>
+	 * } The built tab.
+	 */
+	public function build(): array {
+		$tab = [
+			'target'   => $this->target,
+			'class'    => $this->class,
+			'label'    => $this->label,
+			'id'       => $this->id,
+			'template' => $this->template,
+			'args'     => $this->args,
+		];
+
+		/**
+		 * Filter a specific Help Hub tab.
+		 *
+		 * @since TBD
+		 *
+		 * @param array  $tab The tab data.
+		 * @param string $id  The tab ID.
+		 */
+		$tab = apply_filters( "tec_help_hub_tab_{$this->id}", $tab, $this->id );
```

Comment:
If we need an array here, we should cast the value.
i.e. `$tab = (array) apply_filters(...)`